docs/reference/migration/migrate_2_0.asciidoc

```diff
@@ -138,6 +138,9 @@ In addition some query builders have been removed or renamed:
 * `filtered(...)` removed. Use `filteredQuery(...)` instead.
 * `inQuery(...)` removed.
 
+The methods IndicesQueryBuilder#parseNoMatchQuery(String) and IndicesQueryBuilder#noMatchQuery(String) have been removed,
```

Comment:
BTW I'd in favor of keeping IndicesQueryBuilder#noMatchQuery(String), now that we agree to keep MatchNoneQueryBuilder internal.